controllers/quotes.js

```diff
@@ -0,0 +1,215 @@
+var _ = require('lodash'),
+	crmPrivileges = require('../utils/crm_privileges'),
+	modulesConfig = require('../config/modules_config'),
+	async = require('async'),
+	pagination = require('../utils/pagination'),
+	common = require('../utils/common');
+/**
+ * @swagger
+ * resourcePath: /quotes
+ * description: sQcrm Quotes 
+ */
+module.exports = function(app, config) {
+	var apiNamespace = config.apiNamespace,
+		resourceType = 'quotes';
+	
+	// Public Methods
+	return {
+		/**
+		* @swagger
+		* path: /api/v1/quotes
+		* operations:
+		*   -  httpMethod: GET
+		*      summary: Get all the quotes
+		*      notes: Return all the quotes associated with the token
+		*      responseClass: quotes
+		*      nickname: quotes
+		*      parameters:
+		*        - name: Authorization
+		*          description: access token
+		*          paramType: header
+		*          required: true
+		*          dataType: string
+		*          defaultValue: Bearer ACCESS_TOKEN_HERE
+		*        - name: page
+		*          description: page number of results
+		*          paramType: query
+		*          required: false
+		*          dataType: integer
+		*          defaultValue: 1
+		*        - name: limit
+		*          description: the number of results per page
+		*          paramType: query
+		*          required: false
+		*          dataType: integer
+		*          defaultValue: 50
+		*/
+		getAll: function(req, res, next) {
+			var apiBase = req.protocol + '://' + req.get('host') + apiNamespace,
+				apiEndpoint = apiBase + '/' + resourceType + '/',
+				whereClause,
+				page,
+				limit;
+
+			crmPrivileges.userWhereCondition(req, 13, 'quotes', 'quotes_to_grp_rel', true, function(err, whereCond) {
+				if (err) return next(err);
+				whereClause = whereCond;
+			});
+			
+			pagination.parsePagingRequest(req, function(err, pagingReq) {
+				if (err) return next(err);
+				page = pagingReq.page -1; // limit 0,1 in case the page is 1
+				limit = pagingReq.limit;
+			});
+			
+			
+			async.auto({
+				// get the record count
+				recordCount: function(autoCallback) {
+					var query = " select count(*) as tot from `quotes`";
+						query+= " inner join `quotes_address` on `quotes_address`.`idquotes` = `quotes`.`idquotes`";
+						query+= " inner join `quotes_custom_fld` on `quotes_custom_fld`.`idquotes` = `quotes`.`idquotes`";
+						query+= " left join `user` on `user`.`iduser` = `quotes`.`iduser`";
+						query+= " left join `quotes_to_grp_rel` on `quotes_to_grp_rel`.`idquotes` = `quotes`.`idquotes`";
+						query+= " left join `group` on `group`.`idgroup` = `quotes_to_grp_rel`.`idgroup`";
+						query+= " where `quotes`.`deleted` = 0";
+						query+= whereClause;
+					
+					app.models.quotes
+					.query(query, function(err, quotesCount) {
+						if (err) return autoCallback(err);
+						
+						return autoCallback(null, quotesCount);
+					});
+				},
+				// get the quote prifix
+				getQuotePrifix: ['recordCount', function(result, autoCallback) {
```

Comment:
Looks like this is never used elsewhere, its supposed to be used for the quote number prefix, but the implementation is missing.